### ISSUE TITLE
Blazor UI: Blazorise 0.9.3: Unhandled Exception

### ISSUE THREAD

**Issue by 274188A** [abpframework/abp#7997]
Version: 4.2.2
UI: Blazor

Blazor Project Refs:

```
<PackageReference Include="Blazorise.Bootstrap" Version="0.9.3" />
<PackageReference Include="Blazorise.Icons.FontAwesome" Version="0.9.3" />
```

**Error**

>Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Could not resolve type with token 0100009e from typeref (expected class 'Blazorise.ModalBackdrop' in assembly 'Blazorise, Version=0.9.3.0, Culture=neutral, PublicKeyToken=null')

**Comment by 274188A** [comment]
https://github.com/stsrki/Blazorise/blob/0b50988ea6b6d1535ba8f49f87b600b495507f80/docs/_posts/2021-03-03-release-notes-093.md

**Comment by 274188A** [comment]
still getting same error after removing the offending tags.  cleaned browser cache etc.

**Comment by realLiangshiwei** [comment]
Did you try `dotnet clean & dotnet build`  `.blazor` project

**Comment by 274188A** [comment]
yes of course

**Comment by realLiangshiwei** [comment]
* Steps needed to **reproduce** the problem.

**Comment by 274188A** [comment]
ok, found the problem.  Only works if the `preview` version is used.

The `0.9.3` version fails
The `0.9.3-preview6` works

Note that even running `ABP new myproj -ui blazor` generates `0.9.3-preview6` (why preview on brand new proj?)

So in summary my problem is fix although why does it not work on the release version (ie 0.9.3)?

**Comment by stsrki** [comment]
@274188A you need to follow the migration guide https://blazorise.com/news/release-notes/093/#migration-

`<ModalBackdrop>` is obsolete and should be removed from all razor files.

**Comment by 274188A** [comment]
@stsrki Yes done that, see my previous comment. Problem is that preview 6 is being used for new projects using latest CLI

Follow steps below to reproduce (using CLI and Blazor UI)

**Comment by 274188A** [comment]
ok so to **Reproduce**:
use the CLI

`ABP new testproject -ui blazor`

open up the Blazor project and inspect the project xml.
![image](https://user-images.githubusercontent.com/40014477/110454450-32f88e80-8102-11eb-8943-22cb03892bd2.png)


notice the version nbr for Blazorise

also note that this version is not the latest:

![image](https://user-images.githubusercontent.com/40014477/110454663-78b55700-8102-11eb-8806-0e75642a1312.png)

I suspect something is going amiss in the release pipeline

**Comment by 274188A** [comment]
now if one bumps up to v0.9.3 you will get the following:

![image](https://user-images.githubusercontent.com/40014477/110455381-56700900-8103-11eb-9c2d-0455b02cd005.png)

[All reproducible using a brand new project with the CLI]

**Comment by totpero** [comment]
Is the same issue reported here: https://github.com/abpframework/abp/issues/7946

**Comment by 274188A** [comment]
@totpero - no I believe it is a different issue.
CC: @stsrki  

 The ABP CLI v 4.2.2 generates a brand new blazor project which has the following refs:

    <PackageReference Include="Blazorise.Bootstrap" Version="0.9.3-preview6" />
    <PackageReference Include="Blazorise.Icons.FontAwesome" Version="0.9.3-preview6" />

If you attempt to change the refs to use the so called "stable" version (`9.3.1`) you will see the error pop up as described above.

I simply can't describe it any simpler than this.

Obviously the builds for 9.3.1 still have refs to the `Modalbackdrop` and so since not allowed - the exception is thrown at run-time.

**Comment by stsrki** [comment]
How the CLI generates the PackageReference? I replaced all the references in `.csproj` files to 0.9.3 final.

**Comment by ilkayilknur** [comment]
@stsrki The first preview release will be using the 0.9.3 final packages. When a user creates a new project using the CLI or suite, projects are created with preview packages. I think they are using templates in the `rel-`  branches.

**Comment by 274188A** [comment]
i'm confused as I'm using the CLI tool with 4.2.2 - it produces refs with `0.9.3-preview6` - is it my environment somehow?

**Comment by ilkayilknur** [comment]
Nothing is wrong with your environment. See the template that the CLI and suite use. https://github.com/abpframework/abp/blob/rel-4.2/templates/app/aspnet-core/src/MyCompanyName.MyProjectName.Blazor/MyCompanyName.MyProjectName.Blazor.csproj

I think the problem is Blazorise 0.9.3-final is not backward compatible with 0.9.3-preview6. @stsrki

**Comment by 274188A** [comment]
i thought i was going mad - ok good to know!

**Comment by stsrki** [comment]
@ilkayilknur Yes, that is the case. ModalBackdrop was removed sometimes around preview9 I think.

**Comment by 274188A** [comment]
ok - still, if i remove all the backdrops, upgrade to 0.9.3.1 i still have the problem.  i even cleaned out .vs file, obj, bin

**Comment by ilkayilknur** [comment]
Yes, That's because ABP uses the `ModalBackdrop` internally.

**Comment by 274188A** [comment]
right, so i should wait until that goes away ie 4.2.3?

**Comment by ilkayilknur** [comment]
Yes, that's the safest option, I think.

**Comment by 274188A** [comment]
thanks for getting to the bottom of this @ilkayilknur

**Comment by 274188A** [comment]
reopening until resolved - similar reported in #8149

**Comment by ilkayilknur** [comment]
I don't think that we could take extra action on this issue. It's best to wait for the new preview version. I'm closing the issue.
